t/unit/backends/test_base.py

```diff
@@ -198,6 +203,21 @@ def test_json_exception_arguments(self):
         y = self.b.exception_to_python(x)
         assert isinstance(y, Exception)
 
+    def test_json_exception_nested(self):
+        if sys.version_info < (3, 3):
```

Comment:
next version support py2.7 and 3.5+

Reply:
this is needed for 2.7, so I think this is still needed

Reply:
what about if sys.version_info <(3.5) and (2.7): ?

Reply:
The needed [__qualname__ attribute](https://docs.python.org/3/library/stdtypes.html#definition.__qualname__) was introduced in 3.3, so it's not present in 2.7. I'm not sure what are you trying to address. The condition has to be there, the change can not work in Python older than 3.3, including 2.7. I think it's better to clearly specify version than hiding that into only 2.7 case.

Reply:
so it is a breaking change? Should we defer this to celery 5.x which will be python 3.6+ only?

Reply:
On 2.7 it will behave exactly as before. On 3.3+ it serializes full class name (taken from `__qualname__`) and is able to deserialize such classes.